### PR TITLE
Update README.md to accomodate latest changes

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Update README.md to accomodate latest changes

### Issues Resolved
Closes https://github.com/opensearch-project/spring-data-opensearch/issues/14

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -205,6 +205,35 @@ If you'd rather like the latest snapshots of the upcoming major version, use our
...
+dependencies {
+  ...
+  implementation "org.opensearch.client:spring-data-opensearch:${version}-SNAPSHOT"
+  ...
+}
+
+repositories {
+  ...
+  maven {
+    url = "https://aws.oss.sonatype.org/content/repositories/snapshots/"
```
Comment (dlvenable) [inline README.md:231]: I don't think the snapshots are available yet. Perhaps it is better to clarify that users need to use `mavenLocal()` for the time being.

Comment (reta) [reply README.md:231]: Thanks @dlvenable, I will send the pull request for publishing shapshots today, they should be available shortly

Comment (reta) [reply README.md:231]: https://github.com/opensearch-project/spring-data-opensearch/pull/46

Comment (reta) [reply README.md:231]: @dlvenable are you fine with the change? thanks!

Comment (dlvenable) [reply README.md:231]: Personally I'd rather add the snapshots after they are in. That way anybody checking out the repository is not lead astray. Would you mind adding the snapshot info after they they are deploying?

Comment (reta) [reply README.md:231]: There are already there in the README.md, I am just adding Gradle section

Comment (dlvenable) [reply README.md:231]: That's a fair point on them already being there.

**Review by dlvenable — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (reta) [reply README.md:231]: Thanks @dlvenable !

**Pull request merged**